docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,513 @@
+
+ ...
+}
+--
+
+If you're using repositories, then you will want to extend `PanacheMongoRepositoryBase` instead of `PanacheMongoRepository`
+and specify your ID type as an extra type parameter:
+
+[source,java]
+--
+@ApplicationScoped
+public class PersonRepository implements PanacheMongoRepositoryBase<Person,Integer> {
+
+ ...
+}
+--
+
+[NOTE]
+====
+When using `ObjectId`, MongoDB will automatically provide a value for you, but if you use a custom field type,
+you need to provides the value by yourself.
+====
+
+`ObjectId` can be difficult to use if you want to expose its value in your rest service.
+So we create JsonB providers to serialize/deserialize them as String.
+These providers are not registered by default so you need to register them with a JAXRS `ContextResolver`:
+
+[source,java]
+--
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+import javax.json.bind.JsonbConfig;
+import javax.ws.rs.ext.ContextResolver;
+import javax.ws.rs.ext.Provider;
+
+import io.quarkus.mongodb.panache.jsonb.ObjectIdDeserializer;
+import io.quarkus.mongodb.panache.jsonb.ObjectIdSerializer;
+
+@Provider
+public class QuarkusJsonbContextResolver implements ContextResolver<Jsonb> {
+
+    public Jsonb getContext(Class clazz) {
+        JsonbConfig config = new JsonbConfig();
+        config.withSerializers(new ObjectIdSerializer()).withDeserializers(new ObjectIdDeserializer());
+        return JsonbBuilder.create(config);
+    }
+
+}
+--
+
+== How and why we simplify MongoDB API
+
+When it comes to writing MongoDB entities, there are a number of annoying things that users have grown used to
+reluctantly deal with, such as:
+
+- Duplicating ID logic: most entities need an ID, most people don't care how it's set, because it's not really
+relevant to your model.
+- Dumb getters and setters: since Java lacks support for properties in the language, we have to create fields,
+then generate getters and setters for those fields, even if they don't actually do anything more than read/write
+the fields.
+- Traditional EE patterns advise to split entity definition (the model) from the operations you can do on them
+(DAOs, Repositories), but really that requires an unnatural split between the state and its operations even though
+we would never do something like that for regular objects in the Object Oriented architecture, where state and methods are in the same class. Moreover, this requires two classes per entity, and requires injection of the DAO or Repository where you need to do entity operations, which breaks your edit flow and requires you to get out of the code you're writing to set up an injection point before coming back to use it.
+- MongoDB queries are super powerful, but overly verbose for common operations, requiring you to write queries even
+when you don't need all the parts.
+- MongoDB queries are JSON based, so you will need some String manipulation or using the `Document` type and it will needs a lot of boilerplate code.
```

Comment:
I don't think the first reasons fly for MongoDB. They are more related to traditional JPA development than to the MongoDB API.

Reply:
I can rephrase to
> Even if MongoDB queries are not too verbose, PanacheQL go one step further by allowing to not write any queries for basic use cases.
This is a reference to find("name", name) shortcut query.